homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -73,15 +80,24 @@ async def async_added_to_hass(self):
             self._pvpc_data.state = state.state
 
         # Update 'state' value in hour changes
-        self._hourly_tracker = async_track_time_change(
-            self.hass, self.update_current_price, second=[0], minute=[0]
+        self.async_on_remove(
+            async_track_time_change(
+                self.hass, self.update_current_price, second=[0], minute=[0]
+            )
         )
         # Update prices at random time, 2 times/hour (don't want to upset API)
         random_minute = randint(1, 29)
         mins_update = [random_minute, random_minute + 30]
-        self._price_tracker = async_track_time_change(
-            self.hass, self.async_update_prices, second=[0], minute=mins_update
+        self.async_on_remove(
+            async_track_time_change(
+                self.hass, self.async_update_prices, second=[0], minute=mins_update
+            )
         )
+        # Register callback for when config entry is updated.
+        self.async_on_remove(
+            self._config_entry.add_update_listener(options_update_listener)
```

Comment:
Registering the entry update listener should be done in `async_setup_entry` in the init module.